using System;

class ValidationException : Exception {}

class Post
{
    String content;

    public DateTime ScheduledAt { get; set; }
    public String Url { get; set; }

    public Post(String content)
    {
        this.content = content;
    }

    public void Validate()
    {
        if (String.IsNullOrEmpty(this.content)
            throw new ValidationException();
    }

    public String GetSlug()
    {
        return this.content.Substring(0, 5);
    }

    private bool CSharpMagic()
    {
        return true;
    }
}

class Scheduler
{
    public void PublishOrSchedule(Post post)
    {
        post.Url = String.Format("https://www.example.com/{0}", post.GetSlug());

        if (CSharpMagic())
            post.ScheduledAt = new DateTime();
    }

    private bool CSharpMagic() {
        return false;
    }
}

class PostHandler
{
    Scheduler scheduler;

    public PostHandler(Scheduler scheduler)
    {
        this.scheduler = scheduler;
    }

    public void Execute(String content)
    {
        var post = new Post(content);
        post.Validate();

        this.scheduler.PublishOrSchedule(post);

        if (null != post.ScheduledAt) {
            Console.WriteLine("Published: {0}", post.Url);
        } else {
            Console.WriteLine("Scheduled: {0) at {1:dd/MM}", post.Url, post.ScheduledAt);
        }
    }
}

public class Program
{
    public static void Main()
    {
        new PostHandler(new Scheduler()).Execute("");
    }
}
